packages/next/src/build/webpack/plugins/next-types-plugin.ts

```diff
@@ -272,7 +289,6 @@ declare namespace __next_route_internal_types__ {
   }
 
   type RouteImpl<T> =
-    | StaticRoutes
```

Comment:
I should have added a comment, but this keeps autocompletion working for static routes according to my test in VS Code. Unfortunately it can't support autocompletion for template literals still.

Reply:
I've readded this part :)